### PR TITLE
Adjust permissions of unix socket

### LINKED ISSUE #8358: [Feature request] UNIX socket permissions
Version 4.3 beta 1.
Please add an option to explicitly set UNIX socket permissions.
Thank you.

### PR DESCRIPTION
**Pull request by corny**
closes #8358

### CODE REVIEW

File: pkg/api/http_server.go [hunk 1]
```diff
@@ -74,12 +74,17 @@ func (hs *HttpServer) Start(ctx context.Context) error {
...
 		}
 	case setting.SOCKET:
-		ln, err := net.Listen("unix", setting.SocketPath)
+		ln, err := net.ListenUnix("unix", &net.UnixAddr{Name: setting.SocketPath, Net: "unix"})
 		if err != nil {
 			hs.log.Debug("server was shutdown gracefully")
 			return nil
 		}
 
+		ln.SetUnlinkOnClose(true)
```
Comment (xlson) [inline pkg/api/http_server.go]: Could you remove this line? `true` is the default for unlinkOnClose when the socket is created via ListenUnix.

Comment (xlson) [inline pkg/api/http_server.go]: Could you remove this line? true is the default for unlinkOnClose when the socket is created via net.ListenUnix.

**Comment by corny** [comment]
Oh great, I removed `ln.SetUnlinkOnClose(true)`.

**Review by xlson — CHANGES_REQUESTED** [review]
Thanks for contributing. If you remove line 83 the PR can be merged.

**Comment by xlson** [comment]
@corny Awesome. Thanks!

**Review by xlson — APPROVED** [review]

**Comment by corny** [comment]
Thanks!

**Pull request merged**